### PR TITLE
feat(provider): subscribe to new blocks if possible in heartbeat

### PR DESCRIPTION
**Pull request by DaniPopes**
Ref https://github.com/alloy-rs/alloy/issues/1318

### CODE REVIEW

File: crates/rpc-client/src/builtin.rs [hunk 1]
```diff
@@ -62,17 +62,15 @@ impl BuiltInConnectionString {
...
+                ))
+            }
 
             // hyper is enabled, reqwest is not
             #[cfg(feature = "hyper")]
-            Self::Http(_) => Err(TransportErrorKind::custom_str(
-                "hyper not supported by BuiltinConnectionString. Please instantiate a hyper client manually",
+            Self::Http(url) => Ok(alloy_transport::Transport::boxed(
+                alloy_transport_http::HyperTransport::new_hyper(url.clone()),
             )),
```
Comment (DaniPopes) [inline crates/rpc-client/src/builtin.rs:74]: Why was this not enabled? cc @yash-atreya

File: crates/rpc-client/src/poller.rs
Comment (DaniPopes) [inline crates/rpc-client/src/poller.rs:1]: Best reviewed with "hide whitespace"; this is just moving `async move` to `async fn`

**Review by mattsse — APPROVED** [review]
nice,
ptal @yash-atreya

File: crates/rpc-client/src/builtin.rs (hunk 1 above)
Comment (yash-atreya) [reply crates/rpc-client/src/builtin.rs:74]: Sorry, I missed that! None of the tests covered BuiltInConnectionString.connect_boxed()

Addded coverage in this PR https://github.com/alloy-rs/alloy/pull/1331

**Pull request merged**